src/test/java/core/basesyntax/service/impl/FruitServiceImplTest.java

```diff
@@ -0,0 +1,55 @@
+package core.basesyntax.service.impl;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+import core.basesyntax.db.Storage;
+import core.basesyntax.service.FruitService;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+class FruitServiceImplTest {
+    private Storage storage;
+    private FruitService fruitService;
+
+    @BeforeEach
+    void setUp() {
+        storage = new Storage();
+        fruitService = new FruitServiceImpl(storage);
+    }
+
+    @AfterEach
+    void tearDown() {
+        storage.clear();
+    }
+
+    @Test
+    void save_validData_isOk() {
+        String nameFruit = "banana";
+        int quantity = 25;
+        fruitService.save(nameFruit,quantity);
+        int actual = storage.getFruitQuantity("banana");
+        assertEquals(25, actual);
+
+    }
+
+    @Test
+    void save_addFruitWithZeroQuantity_isOk() {
+        String nameFruit = "banana";

```

Comment:
reused String variable. I'd say there should be a proper constant for it, but c'mon, just change it to "potato" or something))